### PR TITLE
FUL-3056: Protocol checker

### PR DESCRIPTION
**Pull request by bsudy**

### CODE REVIEW

File: ssllabs-junit/src/main/java/io/beekeeper/ssllabs/junit/HostGradeMatcher.java [hunk 1]
```diff
@@ -23,6 +22,18 @@
 
     private final static List<String> GRADES = Arrays.asList("A+", "A", "A-", "B", "C", "D", "E", "F");
     
+    /** Checks if SSLLabs rating is at least <b>A</b> grade. */
+    public final static Matcher<Host> hasAtLeastGradeA = hasAtLeastGrade("A");
```
Comment (Wattos) [inline ssllabs-junit/src/main/java/io/beekeeper/ssllabs/junit/HostGradeMatcher.java]: Should These not be UPPER_CASE as `"constants"`

File: ssllabs-junit/src/main/java/io/beekeeper/ssllabs/junit/ProtocolMatcher.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+		public boolean equals(Object obj) {
+			if (this == obj)
+				return true;
+			if (obj == null)
+				return false;
+			if (getClass() != obj.getClass())
+				return false;
+			ProtocolMatcher.Protocol other = (ProtocolMatcher.Protocol) obj;
+			if (name == null) {
+				if (other.name != null)
```
Comment (Wattos) [inline ssllabs-junit/src/main/java/io/beekeeper/ssllabs/junit/ProtocolMatcher.java]: please always use {} for if statements, even single line

File: ssllabs-junit/src/main/java/io/beekeeper/ssllabs/junit/HostGradeMatcher.java (hunk 1 above)
Comment (bsudy) [inline ssllabs-junit/src/main/java/io/beekeeper/ssllabs/junit/HostGradeMatcher.java]: As I use it in a DSL, I prefer CamelCase... it looks more smooth with the API.

File: ssllabs-junit/src/main/java/io/beekeeper/ssllabs/junit/ProtocolMatcher.java (hunk 2 above)
Comment (bsudy) [inline ssllabs-junit/src/main/java/io/beekeeper/ssllabs/junit/ProtocolMatcher.java]: Shit.... auto generated equals :facepalm:

**Pull request merged**